### ISSUE TITLE
Unable to update `requestBody` without using a deprecated record setter

### ISSUE THREAD

**Issue by Profpatsch** [yesodweb/wai#849]
https://hackage.haskell.org/package/wai-3.2.2.1/docs/Network-Wai.html#v:requestBody is deprecated, thus there is no way of using it as a setter without using a deprecated symbol.

I used the following workaround module, I suggest we add this setter function:

```
{-# OPTIONS_GHC -Wno-deprecations #-}
{-# OPTIONS_GHC -Wno-name-shadowing #-}

module Wai where

import Data.ByteString (ByteString)
import Network.Wai (Request (requestBody))

-- | Workaround to un-deprecate the setter. see 'requestBody'.
setRequestBodyChunk :: IO ByteString -> Request -> Request
setRequestBodyChunk requestBody r =
  r {requestBody = requestBody}
```

**Comment by snoyberg** [comment]
PR welcome

**Comment by Vlix** [comment]
Would `setRequestBodySource` be a more appropriate name?

**Comment by kazu-yamamoto** [comment]
I guess that @Profpatsch  chose `setRequestBodyChunk` as a partner of `getRequestBodyChunk`.

**Comment by Vlix** [comment]
Yeah, but it's not setting a chunk... so would it maybe be better to then just name it `setRequestBody`?

It's an easy addition, so I think I might pick this up. (maybe bundle it together with #710 )

**Comment by Vlix** [comment]
Hmmm, ok I can see your point. I guess the most important thing is to emphatically add documentation that the `IO ByteString` MUST provide data until the request body is completely produced, after which it MUST return `""`. So for example that someone shouldn't `setRequestBodyChunk (pure bytestring) request`, because that will cause loops or memory leaks if someone expects it to produce chunks. :thinking: 

Should we move this discussion over to #949 or should @dbeacham just keep an eye on this discussion? :sweat_smile:

**Comment by dbeacham** [comment]
> In my understanding, this `IO ByteString` can be called multiple times and returns "" finally.
> I call it chunks.

I think `setRequestBodyChunks` is a reasonable name to complement `getRequestBodyChunk` given how they are both expected to behave.

I can also add some docs to mirror the information given in `requestBody` and `getRequestBodyChunk` that it should return `""` once the body has been fully produced.

**Comment by Vlix** [comment]
Implemented in #949
